base/src/main/java/io/spine/validate/ValidationRulesMap.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.validate;
+
+import com.google.common.collect.ImmutableBiMap;
+import com.google.protobuf.Descriptors.Descriptor;
+import com.google.protobuf.Descriptors.FieldDescriptor;
+import io.spine.type.TypeName;
+
+import java.util.Properties;
+import java.util.Set;
+
+import static com.google.protobuf.Descriptors.FieldDescriptor.JavaType.MESSAGE;
+import static io.spine.util.Exceptions.newIllegalStateException;
+import static io.spine.util.PropertyFiles.loadAllProperties;
+
+/**
+ * A map from a validation rule descriptor to the field descriptor of the rule target.
+ *
+ * @author Dmytro Grankin
+ */
+class ValidationRulesMap {
+
+    /**
+     * A path to the file, which contains validation rules and their target fields paths.
+     */
+    @SuppressWarnings("DuplicateStringLiteralInspection") // To avoid the undesirable dependency
+                                                          // on the model compiler.
+    private static final String PROPS_FILE_NAME = "validation_rules.properties";
+    private static final String PROTO_TYPE_SEPARATOR = ".";
+
+    private static final ImmutableBiMap<Descriptor, FieldDescriptor> rules = buildValidationRulesMap();
+
+    private ValidationRulesMap() {
+        // Prevent instantiation of this class.
+    }
+
+    /**
+     * Obtains validation rules map instance.
+     *
+     * @return the immutable map
+     */
+    @SuppressWarnings("ReturnOfCollectionOrArrayField") // As return value is an immutable collection.
+    static ImmutableBiMap<Descriptor, FieldDescriptor> getInstance() {
+        return rules;
+    }
+
+    private static ImmutableBiMap<Descriptor, FieldDescriptor> buildValidationRulesMap() {
```

Comment:
The method name is too long and specific. Please simplify. Perhaps, `build` would work.